utils/common/src/main/kotlin/DirectoryStash.kt

```diff
@@ -32,24 +32,27 @@ import kotlin.io.path.moveTo
 import org.apache.logging.log4j.kotlin.Logging
 
 /**
- * A convenience function that stashes directories using a [DirectoryStash] instance.
+ * A convenience function that stashes [directories] to the [stashParentDir] using a [DirectoryStash] instance.
  */
-fun stashDirectories(vararg directories: File): Closeable = DirectoryStash(setOf(*directories))
+fun stashDirectories(
+    vararg directories: File,
+    stashParentDir: File = createTempDirectory("stashParent").toFile()
```

Comment:
Is the new stash location of `stashParentDir` guaranteed to be on same file system?

Reply:
No. I've put some effort into finding out whether there's a reliable way to do that, but I haven't found any. So the best we can probably do is to document that requirement.

Reply:
The new approach now again ensures temporary directories to be created as siblings on the same filesystem.